File: platforms/ios/WeexEros/WeexEros/Component/Weex.m
//
//  Weex.m
//  WeexEros
//
//  Created by chen century on 2018/2/23.
//  Copyright © 2018年 benmu. All rights reserved.
//

#import "Weex.h"
#import "pgyupdate.h"
#import "RyIm.h"
#import "Tjchart.h"
#import <RongIMKit/RongIMKit.h>
#import "RyListView.h"
#import "PlayerView.h"
#import "EzplayView.h"

@implementation Weex
+(void)initWeex:(NSString *)group{
    [WXSDKEngine initSDKEnvironment];
    [[RCIM sharedRCIM] initWithAppKey:@"c9kqb3rdcyhqj"];
    
    [self registerModules];
    [self registerComponents];
}

+ (void)registerModules
{
    NSDictionary *modules = @{
                              @"pgyupdate": NSStringFromClass([pgyupdate class]),
                              @"rongyun": NSStringFromClass([RyIm class])
                              };
    
    for (NSString *moduleName in modules.allKeys) {
        [WXSDKEngine registerModule:moduleName withClass:NSClassFromString([modules valueForKey:moduleName])];
    }
}
+ (void)registerComponents
{
    NSDictionary *components = @{
                                @"rylist": NSStringFromClass([RyListView class]),
                                @"tjEcharts": NSStringFromClass([Tjchart class]),
                                @"tj-video": NSStringFromClass([PlayerView class]),
                                @"tj-ysy-video": NSStringFromClass([EzplayView class])
                                 };
    for (NSString *componentName in components) {
        [WXSDKEngine registerComponent:componentName withClass:NSClassFromString([components valueForKey:componentName])];
    }
}

@end

File: platforms/android/WeexFrameworkWrapper/app/src/main/java/com/century/tianjiapp/Weex.java
package com.century.tianjiapp;

import com.century.tianjiapp.RyIm.RyListView;
import com.century.tianjiapp.RyIm.RyRegist;
import com.century.tianjiapp.Video.PlayerView;
import com.century.tianjiapp.Video.ysyView;
import com.century.tianjiapp.cpmpontes.TJehcarts;
import com.century.tianjiapp.pgy.pgyupdate;
import com.taobao.weex.WXSDKEngine;
import com.taobao.weex.common.WXException;


/**
 * Created by chencentury on 2018/2/23.
 */

public class Weex {
    private static Weex instance = null;
    private Weex(){}

    public static Weex getInstance(){
        if(instance == null){
            instance = new Weex();
        }
        instance.init();
        return instance;
    }

    //Context 上下文
    // Application  app的类
    public void init(){
        try {
            WXSDKEngine.registerModule("pgyupdate", pgyupdate.class);
            WXSDKEngine.registerModule("rongyun", RyRegist.class);

            WXSDKEngine.registerComponent("rylist", RyListView.class);
            WXSDKEngine.registerComponent("tjEcharts", TJehcarts.class);
            WXSDKEngine.registerComponent("tj-video", PlayerView.class);
            WXSDKEngine.registerComponent("tj-ysy-video", ysyView.class);
        }catch (WXException e){
            e.printStackTrace();
        }
    }
}
